modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/common/main/infrastructure/OffsetDateTime.kt.mustache

```diff
@@ -0,0 +1,89 @@
+package {{packageName}}.infrastructure
+
+import kotlinx.datetime.*
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+/**
+ * Represents a [LocalDateTime] and the respective [TimeZone] of it.
+ */
+@Serializable(with = OffsetDateTime.Companion::class)
+public data class OffsetDateTime(val dateTime: LocalDateTime, val timeZone: TimeZone) {
+    override fun toString(): String {
+        return if (timeZone == TimeZone.UTC) {
+            "${dateTime}Z"
+        } else {
+            "$dateTime$timeZone"
+        }
+    }
+
+    /**
+     * Converts the [OffsetDateTime] to an [Instant]. This looses the [TimeZone] information, because the date and time
+     * is converted to UTC in the process.
+     */
+    public fun toInstant(): Instant = dateTime.toInstant(timeZone)
+
+    /**
+     * Creates an [OffsetDateTime] from an [Instant] in a given [TimeZone] ([TimeZone.UTC] by default).
+     */
+    public fun fromInstant(instant: Instant, timeZone: TimeZone = TimeZone.UTC): OffsetDateTime = OffsetDateTime(
```

Comment:
This does not use the instance and would make more sense to be declared in the `companion object`.

Reply:
Ehm, yeah this isn't supposed to be on the instance...